### PR TITLE
Eclipse OMR Generative AI (GenAI) Usage Guidelines

### PR DESCRIPTION
**Pull request by 0xdaryl**

### CODE REVIEW

**Comment by 0xdaryl** [comment]
At the very least, I'd like all active committers to read and provide any feedback by Dec. 5.  Input from anyone else in the community is welcome as well.

@mstoodle @vijaysun-omr @dsouzai @jdmpapin @hzongaro @babsingh @knn-k

FYI @pshipton @keithc-ca @tajila @kevindean12

File: doc/GenAIUsageGuidelines.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+“assisted” distinction is necessary because it may not be possible to clearly
+identify just the AI-generated content in the contribution, and it allows for
+human changes to the demarcated contribution over time while preserving the
+meaning of the tags.
+
+### Tag Format
+
+At this time, Eclipse OMR does not specify the format of the tags, except that
+they must not contain any corporate-specific information or other identifiers
+that have no meaning beyond the original contributor.  Information about the
```
Comment (hzongaro) [inline doc/GenAIUsageGuidelines.md:87]: > they must not contain any corporate-specific information or other identifiers
that have no meaning beyond the original contributor.

It's not clear to me what this means.  If my employer, HAL Corp., say, requires me to tag code generated by its AI assistant as `// {begin|end} generated-by-hal-ai` or `// {begin|end} hal-generated-content-is-the-best`, is that corporate-specific information?  Is the idea that the tagging should indicate that the demarcated region was generated by some AI, but that it must not contain a trademark of any sort?

File: doc/GenAIUsageGuidelines.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+Inline tagging of code or documentation generated by coding assistants is not
+mandatory.
+
+If tags are used, they must clearly demarcate the boundaries of GenAI-assisted
+content (e.g., via paired "begin" and "end" code comments). These boundaries
+may span consecutive lines, functions, paragraphs, or entire files. The term
+“assisted” is an important qualifier to describe the enclosed content because
+it may contain a blend of AI-generated content and human contributions. The
+“assisted” distinction is necessary because it may not be possible to clearly
```
Comment (hzongaro) [inline doc/GenAIUsageGuidelines.md:78]: Nit:  the quote characters that surround "begin" and "end" are different from the quote character that surrounds “assisted”

**Review by hzongaro — COMMENTED** [review]
I think this looks very good.  Just one question for clarification.

File: doc/GenAIUsageGuidelines.md (hunk 1 above)
Comment (keithc-ca) [reply doc/GenAIUsageGuidelines.md:87]: My reading of that section is that the tags are free to refer the specific AI assistant which is likely to contain or imply the vendor of that assistant, HAL in your example, which is fine whether you work for HAL or simply use HAL's assistant.

I think the prohibition is against things that have no meaning to anyone who isn't privy to things behind the corporate-specific tag (e.g. references to issues or other proprietary information accessible only to employees of a certain corporation). Also inappropriate would be any subjective commentary like "is-the-best" in your example.

File: doc/GenAIUsageGuidelines.md (hunk 2 above)
Comment (keithc-ca) [reply doc/GenAIUsageGuidelines.md:78]: I agree that quoting should be consistent - my preference would be to use `"` consistently (unicode character 0x22, not the opening and closing quotes that appear around "assisted").

**Comment by keithc-ca** [comment]
This looks good to me, assuming the "nit" Henry raised is addressed.

File: doc/GenAIUsageGuidelines.md (hunk 1 above)
Comment (hzongaro) [reply doc/GenAIUsageGuidelines.md:87]: That makes sense.  If that's what was intended, I think it would be worth clarifying.

**Review by vijaysun-omr — APPROVED** [review]

**Review by knn-k — APPROVED** [review]

**Review by mstoodle — APPROVED** [review]

**Review by babsingh — APPROVED** [review]

**Review by dsouzai — APPROVED** [review]

File: doc/GenAIUsageGuidelines.md (hunk 2 above)
Comment (0xdaryl) [reply doc/GenAIUsageGuidelines.md:78]: Hm, thanks for spotting that.  I have no idea how that happened.  `"` was my intent, and I'll fix.

File: doc/GenAIUsageGuidelines.md (hunk 1 above)
Comment (0xdaryl) [reply doc/GenAIUsageGuidelines.md]: Keith's interpretation is correct.  I (hopefully) added some clarification to the text in the forced push.

**Review by hzongaro — APPROVED** [review]
Looks good.  Thanks!

File: doc/GenAIUsageGuidelines.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+attribute that at the end of the commit message (before any human authorship
+signatures).
+
+```
+Co-authored by <YOUR AI ASSISTANT>
+```
+
+## Tagging
+
+Some contributors’ employers may require their AI-generated code and
```
Comment (keithc-ca) [inline doc/GenAIUsageGuidelines.md:62]: Suggest using a plain ASCII apostrophe (unicode 0x27) instead of `’`.

Comment (0xdaryl) [reply doc/GenAIUsageGuidelines.md]: Fixed.

**Comment by 0xdaryl** [comment]
Apologies for the delay. I was considering some private input on this. I believe all reviewer comments have been addressed. The most significant change with the recent push is that the GenAI guidelines are no longer in a separate document but part of the contributor guidelines. The wording is identical from what you reviewed before.

I will ask @mstoodle to merge today unless anyone has any final feedback.

**Comment by mstoodle** [comment]
I'll merge at 2pm-ish unless there are new comments to address.

**Pull request merged**